### PR TITLE
Fix trigger_push behaviour

### PR DESCRIPTION
**Pull request by tilgep**
Change trigger_push back to set base velocity instead of adding to your abs velocity every tick.

### CODE REVIEW

File: src/detours.cpp [hunk 1]
```diff
@@ -24,6 +25,78 @@ DECLARE_DETOUR(UTIL_SayText2Filter, Detour_UTIL_SayText2Filter, &modules::server
...
+
+	// if push up and FL_ONGROUND
+	if (vecPush.z > 0 && (flags & 1))
+	{
+		addresses::SetGroundEntity(pOther, nullptr);
+		Vector origin = pOther->GetAbsOrigin();
+		origin.z += 1.0f;
+
+		//CBaseEntity::Teleport
+		CALL_VIRTUAL(void, 148, pOther, origin, nullptr, nullptr);
```
Comment (xen-000) [inline src/detours.cpp:91]: I suggest you add a function in Z_CBaseEntity for this.

File: src/detours.cpp [hunk 2]
```diff
@@ -24,6 +25,78 @@ DECLARE_DETOUR(UTIL_SayText2Filter, Detour_UTIL_SayText2Filter, &modules::server
...
+	if (movetype == MOVETYPE_VPHYSICS)
+	{
+		TriggerPush_Touch(pPush, pOther);
+		return;
+	}
+
+	Z_CBaseEntity* pPushEnt = (Z_CBaseEntity*)pPush;
+
+	// SF_TRIG_PUSH_ONCE is handled fine already
+	if (pPushEnt->m_spawnflags() & 0x80)
```
Comment (xen-000) [inline src/detours.cpp:44]: Even if just used here, I think there should be a define for the flags

File: src/detours.cpp [hunk 3]
```diff
@@ -24,6 +25,78 @@ DECLARE_DETOUR(UTIL_SayText2Filter, Detour_UTIL_SayText2Filter, &modules::server
...
+	// i had issues with vectorrotate on linux so i did it here
+	vecAbsDir.x = pushDir.x * mat[0][0] + pushDir.y * mat[0][1] + pushDir.z * mat[0][2];
+	vecAbsDir.y = pushDir.x * mat[1][0] + pushDir.y * mat[1][1] + pushDir.z * mat[1][2];
+	vecAbsDir.z = pushDir.x * mat[2][0] + pushDir.y * mat[2][1] + pushDir.z * mat[2][2];
+
+	Vector vecPush = vecAbsDir * pPush->m_flPushSpeed();
+
+	uint32 flags = pOther->m_fFlags();
+
+	if (flags & (0x800000))
```
Comment (xen-000) [inline src/detours.cpp:78]: ditto

**Comment by Poggicek** [comment]
In first glance looks ok except for the few magic numbers, didn't test the functionality yet.

File: src/detours.cpp [hunk 4]
```diff
@@ -24,6 +25,78 @@ DECLARE_DETOUR(UTIL_SayText2Filter, Detour_UTIL_SayText2Filter, &modules::server
...
+		origin.z += 1.0f;
+
+		//CBaseEntity::Teleport
+		CALL_VIRTUAL(void, 148, pOther, origin, nullptr, nullptr);
+	}
+
+	pOther->m_vecBaseVelocity(vecPush);
+
+	// Add FL_BASEVELOCITY flag
+	flags |= (0x800000);
```
Comment (xen-000) [inline src/detours.cpp:97]: I'm guessing the flags defined in the sdk are different to this? Because otherwise you can just put FL_BASEVELOCITY in there.

**Review by xen-000 — CHANGES_REQUESTED** [review]

**Review by xen-000 — APPROVED** [review]

**Pull request merged**